src/DataShare/DataShare/ProviderShareSubscriptions/RevokeAzDataShareSubscriptionAccess.cs

```diff
@@ -0,0 +1,126 @@
+    using Microsoft.Azure.Management.DataShare.Models;
+    using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+    using Microsoft.Azure.PowerShell.Cmdlets.DataShare.Extensions;
+    using Microsoft.Azure.PowerShell.Cmdlets.DataShare.Models;
+
+    /// <summary>
+    /// Defines Revoke-AzDataShareSubscriptionAccess cmdlets.
+    /// </summary>
+    [Cmdlet("Revoke", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "DataShareSubscriptionAccess", DefaultParameterSetName = ParameterSetNames.FieldsParameterSet, SupportsShouldProcess = true),
+     OutputType(typeof(PSDataShareProviderShareSubscription))]
+    public class RevokeAzDataShareSubscriptionAccess : AzureDataShareCmdletBase
+    {
+
+        /// <summary>
+        /// The resource group name of the azure data share account.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource group name of the azure data share account",
+            ParameterSetName = ParameterSetNames.FieldsParameterSet)]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Name of azure data share account.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Azure data share account name",
+            ParameterSetName = ParameterSetNames.FieldsParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(ResourceTypes.Account, "ResourceGroupName")]
+        public string AccountName { get; set; }
+
+        /// <summary>
+        /// Name of azure data share.
+        /// </summary>
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Azure data share name",
+            ParameterSetName = ParameterSetNames.FieldsParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [ResourceNameCompleter(ResourceTypes.Share, "ResourceGroupName", "AccountName")]
+        public string ShareName { get; set; }
+
+        /// <summary>
+        /// The share subscription id of the provider share subscription.
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The share subscription id of the provider share subscription",
+            ParameterSetName = ParameterSetNames.FieldsParameterSet)]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The share subscription id of the provider share subscription",
+            ParameterSetName = ParameterSetNames.ResourceIdParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string ShareSubscriptionId { get; set; }
```

Comment:
Why is there ShareSubscriptionID + ResourceId in thsi cmdlet, but only ShareSUbscriptionId in the associated 'Grant' cmdlet?

Reply:
Grant comdlet also has resourceId and share subscriptionId in it